seisflows/optimize/gradient.py

```diff
@@ -167,9 +172,15 @@ def check(self):
         if self.step_len_max is not None:
             assert 0. < self.step_len_max, \
                 f"optimize.step_len_max must be >= 0."
+        if self.step_len_min is not None:
+            assert 0. < self.step_len_min, \
```

Comment:
You are using an f-string but there is no variable in the string.

Or you can also add "..., but got {self.step_len_min}." To make the error message better ;)

Reply:
Thanks, I sometimes leave the f on a string, I imagine any performance hit for evaluating an f-string is negligible.